justfile

```diff
@@ -473,3 +473,19 @@ mdbook *args="build":
       --entrypoint /bin/mdbook \
       {{ _doc_env_container }} \
       {{ args }}
+
+# Run a simple build for each separate commit (for "pull_request")
+# or for the HEAD of the branch (other events).
+[script]
+build-sweep start="main":
+    {{ _just_debuggable_ }}
+    set -eu -o pipefail
+    if [ {{_clean}} != "clean" ]; then
+      >&2 echo "can not build-sweep with dirty branch (would risk data loss)"
+      exit 1
+    fi
+    # Get all commits since {{start}}, in chronological order
+    while read -r commit; do
+      git checkout "${commit}" || exit 1
+      { just debug=true cargo +{{rust}} build --locked --profile=dev --target=x86_64-unknown-linux-gnu; } || exit 1
```

Comment:
I'm not convinced that having commit-by-commit checking of build is a good idea, but it is fine to leave it for now, we can drop the requirement if it becomes a problem.  Build is probably fine though, unit test is where it gets hairy.